docs/proposal/2022-11-22-botkube-plugins-integration.md

```diff
@@ -0,0 +1,498 @@
+    * [The `plugins` object syntax](#the-plugins-object-syntax)
+    * [How a given plugins is configured](#how-a-given-plugins-is-configured)
+<!-- TOC -->
+
+## Motivation
+
+This proposal describe how the [plugin system](2022-09-28-botkube-plugin-system.md) can be integrated with the Botkube core. It describes both the configuration syntax changes and expected behavior.
+
+## New syntax
+
+This section describes the necessary changes in the syntax. **It's backward compatible.**
+
+1. A new `plugins` property. It allows to specify a list of repository from where the plugins can be downloaded.
+   ```yaml
+   plugins:
+     cacheDir: "/tmp"
+     repositories:
+       botkube: https://plugins.botkube.io/botkube.yaml
+       huseyinbabal: https://raw.githubusercontent.com/huseyinbabal/botkube-plugins/main/index.json
+   ```
+
+2. The `executors` definition now can refer to the executor plugins specified in a given repository.
+   ```yaml
+   executors:
+     'plugin-based':
+       botkube/kubectl:     # <repo>/<plugin> is syntax for plugin based executors
+         enabled: true      # if not enabled we don't download and start a given plugin
+         version: v1.0.0    # if empty, the latest version is used
+         config:            # plugin specific configuration
+           namespaces:
+             include: ["botkube", "default","ambassador"]
+           commands:
+             verbs: ["get","logs","top"]
+             resources: ["pods","deployments","nodes","configmap"]
+   ```
+
+3. The `sources` definition now can refer to the source plugins specified in a given repository.
+   ```yaml
+   sources:
+     'plugin-based':
+       botkube/kubernetes:
+         enabled: true
+         version: v1.0.0
+         config:
+           recommendations:
+             pod:
+               noLatestImageTag: true
+               labelsSet: true
+   ```
+
+## Use cases
+
+This section describes example configurations that enable the requested use-cases.
+
+### Defining list of plugin repositories and enabling a given plugin
+
+We introduce a new `plugins` object to Botkube configuration syntax. It holds the list of the plugins repositories. Each repository has own name to prevent conflict with the same plugins name, e.g. both provides `kubectl` plugin.
+
+To enable a given plugin you need to specify it under the `executors` or `sources` property. In the first phase, we only allow to enable and disable plugins using Botkube configuration. Later we can introduce `@Botkube enable [executor|source] NAME` and `@Botkube disable [executor|source] NAME`. Disabling and enabling plugins should be possible in the Botkube runtime without restarting it.
+
+**Syntax:**
+
+```yaml
+executors:
+  'plugin-based':
+    botkube/kubectl:     # <repo>/<plugin> is syntax for plugin based executors
+      enabled: true      # if not enabled we don't download and start a given plugin
+      version: v1.0.0    # if empty, the latest version is used.
```

Comment:
This version somehow doesn't fit here to me, as it is more about the plugin metadata, like `{repo}/{plugin-name}` 🤔 What do you think to have something like

```yaml
executors:
  'plugin-based':
    botkube/kubectl@v1.0.0:     # <repo>/<plugin> is syntax for plugin based executors
      enabled: true      # if not enabled we don't download and start a given plugin
      config: {}
```

?

Reply:
I like it 👍

Reply:
When I see version, I expect there might be 2 versions of same plugin at the same time. Could you please elaborate on why we don't  trust the version in dependencies section only and eliminate from here?